sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -557,10 +557,12 @@ std::unique_ptr<RawResponse> WinHttpTransport::Send(Context const& context, Requ
 }
 
 // Read the response from the sent request.
-int64_t Details::WinHttpStream::Read(Context const& context, uint8_t* buffer, int64_t count)
+int64_t Details::WinHttpStream::ReadImplementation(
+    Context const& context,
+    uint8_t* buffer,
+    int64_t count)
 {
-  context.ThrowIfCanceled();
```

Comment:
I am surprised we are removing these. Please leave them or help me understand why we need to get rid of them.

Reply:
Calling `Read()` from a winHttpBodyStream will fisrt call the base body stream `Read` method which makes this cancellation check. Then that method calls this ReadImplementation.  Leaving this check will mean an instant double check for cancellation.

And the removal of the second check below is because we want to do the cancellation check only before I/O calls.  
Calling `WinHttpQueryDataAvailable` is not an I/O because winHTTP will just tell you how much data is there for you without reading any socket/network.
It is not until we call `WinHttpReadData` that we are doing an I/O

Reply:
Yes, and I think now that ReadImplementation should be renamed OnRead because this OnXxx pattern is common in OOP. You see it in .NET a lot; maybe it's common in C++ too (@CaseyCarter ?)

Reply:
> Calling `WinHttpQueryDataAvailable` is not an I/O because winHTTP will just tell you how much data is there for you without reading any socket/network.
It is not until we call `WinHttpReadData` that we are doing an I/O

Are you sure about that? Given some of the potential error return codes and the fact we allow async here, I do think the underlying implementation might do an I/O here.

Reply:
In that case, we could leave the second check

Reply:
In that case, we could leave the second check

Reply:
restored the second canceled() checked